src/Language/PureScript/TypeChecker/Types.hs

```diff
@@ -388,6 +388,10 @@ infer' (IfThenElse cond th el) = do
 infer' (Let ds val) = do
   (ds', val'@(TypedValue _ _ valTy)) <- inferLetBinding [] ds val infer
   return $ TypedValue True (Let ds' val') valTy
+infer' (Proxy ty) = do
+  _ <- kindOfWithScopedVars ty
+  ty' <- introduceSkolemScope <=< replaceAllTypeSynonyms <=< replaceTypeWildcards $ ty
+  return $ TypedValue False (Proxy ty') (ProxyType ty')
```

Comment:
This doesn't matter too much, but I use this boolean to indicate "this should be checked", and then "this _was_ checked", so it'd be good to be consistent is all.